lib/index.js

```diff
@@ -31,17 +47,24 @@ exports = module.exports = function historyApiFallback (req, res, next) {
 
   var parsedUrl = url.parse(req.url);
   if (parsedUrl.pathname.indexOf('.') !== -1) {
-    logger('Not rewriting %s %s because the path includes a dot (.) character.',
-      req.method, req.url);
-    return next();
+    if(!rewriteUrlsWithDots) {
+      logger('Not rewriting %s %s because the path includes a dot (.) character.',
+        req.method, req.url);
+      return next();
+    }
+    else {
+      var fileName = parsedUrl.href.split(parsedUrl.search).join("");
+      var fileExists = fs.existsSync(appBaseDir + fileName);
```

Comment:
Handling `appBaseDir` and checking for existing files looks like a completely separate rule. This may also simplify usage of historyApiFallback in general. Could you please refactor this to a separate option?